containers/src/Kokkos_DualView.hpp

```diff
@@ -275,14 +275,29 @@ class DualView : public ViewTraits<DataType, Properties...> {
            const size_t n5                   = KOKKOS_IMPL_CTOR_DEFAULT_ARG,
            const size_t n6                   = KOKKOS_IMPL_CTOR_DEFAULT_ARG,
            const size_t n7                   = KOKKOS_IMPL_CTOR_DEFAULT_ARG)
-      : modified_flags(t_modified_flags("DualView::modified_flags")),
-        d_view(arg_prop, n0, n1, n2, n3, n4, n5, n6, n7) {
-    // without UVM, host View mirrors
-    if constexpr (Kokkos::Impl::has_type<Impl::WithoutInitializing_t,
-                                         P...>::value)
-      h_view = Kokkos::create_mirror_view(Kokkos::WithoutInitializing, d_view);
-    else
-      h_view = Kokkos::create_mirror_view(d_view);
+      : modified_flags(t_modified_flags("DualView::modified_flags")) {
+    if constexpr (Impl::ViewCtorProp<P...>::sequential_host_init) {
```

Comment:
I think it's time to discuss if we want to allow the dual view constructor to be passed both host and device `arg_prop`.

With this change, it means I cannot build the device view on a given execution space instance.

Reply:
You can but not with `SequentialHostInit` (which didn't work properly before anyway) so this doesn't create any new restrictions.

> I think it's time to discuss if we want to allow the dual view constructor to be passed both host and device arg_prop.

Sure but not here.

Reply:
I was suggesting a constructor that takes 2 `ViewCtorProp`, since we want to build 2 views.

Reply:
That's out-of-scope. The point of this pull request is to make sure we can use `SequentialHostInit` with `DualView` in a reasonable way. Adding another `ViewCtorProp` is orthogonal to that.

Reply:
Let's take that discussion elsewhere

Reply:
By that I agree this is out of scope of this work, please open a separate issue

Reply:
I agree with @romintomasetti that it is important that it works to pass an execution space instance. The current behavior is that if the user passes an execution space instance to the constructor in the constructor properties, this execution space instance is forwarded to the constructor of the device view, but not to the constructor of the host view. So it seems fair to try to maintain this behavior with `SequentialHostInit`.

A solution with two constructor properties looks like a new approach. So I can understand that such a solution would be out of scope.

Perhaps there are other solutions. With the constructor currently in this PR, if the user passes an `arg_prop` with `SequentialHostInit`, that `arg_prop` is forwarded "as is" to the constructor of the host view. And so there can't be a device execution space in that `arg_prop`. Does this suggest that we should not forward that `arg_prop` "as is" to the constructor of the host view? E.g. a solution whereby we'd create a new `arg_prop_h` (without the passed exec, if any) that we forward to the constructor of the host view, and a new `arg_prop_d` (with the passed exec, if any), that we pass to the `create_mirror_view` to create the device view.

Reply:
The status quo (what is currently in develop) is that the SequentialHostInit is essentially incompatible with DualView.
This PR stands on its own.  It enables the most straightforward use cases, bringing basic support.
Romin is asking for more advanced use cases which raises a lot more questions.
My ruling that this is out of scope is not a judgment of the value/importance of advance use cases, just pragmatism.  The current PR is an improvement and does not prevent further extensions.

Reply:
Hi @dalg24. Sorry, it looks like you replied while I was typing my comment. OK to take it elsewhere indeed.